SFFeatureFlag/Classes/Services/FeatureFlagCenter.swift

```diff
@@ -0,0 +1,159 @@
+        if let startDate = criteria.activeInterval?.startDate?.toDate(),
+           startDate.timeIntervalSince1970 > Date().timeIntervalSince1970 {
+            return false
+        }
+        
+        //Check the active interval endDate
+        if let endDate = criteria.activeInterval?.endDate?.toDate(),
+           endDate.timeIntervalSince1970 < Date().timeIntervalSince1970 {
+            return false
+        }
+        
+        //Check the specific conditions
+        if let conditions = criteria.specificConditions, !conditions.isEmpty {
+            for condition in criteria.specificConditions ?? [] {
+                if !isConditionSatisfied(condition, with: conditionsDataSource[condition.key]) {
+                    return false
+                }
+            }
+        }
+        
+        return true
+    }
+    
+    /// Checks whether the passed value satisfies the certain condition
+    func isConditionSatisfied(_ condition: Condition, with value: Any?) -> Bool {
+        guard let value = value else { return false }
+        
+        let validValues = condition.validValues.compactMap {$0.lowercased()}
+        
+        if let valueString = value as? String {
+            let containedInCondition = condition.validValues.contains(valueString)
+            
+            return condition.inverse ? !containedInCondition : containedInCondition
+        } else if let valueArray = (value as? [String])?.compactMap({$0.lowercased()}) {
+            let valuesIntersection = Set(valueArray).intersection(validValues)
+            
+            return condition.inverse ? valuesIntersection.isEmpty : !valuesIntersection.isEmpty
+        } else if let valueArray = value as? [Any] {
+            let mappedArray = valueArray.compactMap {"\($0)"}
+            
+            return isConditionSatisfied(condition, with: mappedArray)
+        }
+        
+        return isConditionSatisfied(condition, with: "\(value)")
+    }
+    
+    ///Determines if the passed version is with the version range
+    func isVersionInRange(version: String, versionRange: VersionRange) -> Bool {
+        if let minVersion = versionRange.min, Double(minVersion) ?? 0.0 > Double(version) ?? 0.0 {
+            return false
+        }
+        
+        if let maxVersion = versionRange.max, Double(maxVersion) ?? 0.0  < Double(version) ?? 0.0 {
+            return false
+        }
+        
+        return true
+    }
+    
+    /// Gets the featurs config data from the remote config service and decodes it to a list of FeatureConfig objects.
+    func getFeaturesConfig() -> [FeatureConfig] {
+        let jsonDecoder = JSONDecoder()
+        jsonDecoder.keyDecodingStrategy = featuresConfigDecodingStrategy
+        
+        if let featuresConfigData = remoteConfigService.configValue(forKey: featuresConfigKey, type: Data.self) {
+            let featuresConfig = try? jsonDecoder.decode([FeatureConfig].self, from: featuresConfigData)
```

Comment:
This could be chained in order to avoid the nil coalescing operator

Reply:
Updated the code with this suggestion